### PR TITLE
Remove all incorrect usages of std::string

### PR DESCRIPTION
**Pull request by LevchenkoS**
Currently, in code present incorrect usage
 of std::string like this:
`std::string("Some string").c_str();`
It causes undefined behavior, and should
be deprecated.

-
Related to: [APPLINK-28458](https://adc.luxoft.com/jira/browse/APPLINK-28458)

@AGaliuzov, @Kozoriz, @LuxoftAKutsan, @VProdanov, @wolfylambova22, please, review.

### CODE REVIEW

File: src/components/media_manager/src/audio/from_mic_to_file_recorder_thread.cc [hunk 1]
```diff
@@ -120,14 +120,16 @@ media_manager::FromMicToFileRecorderThread::Impl::Impl(
 
   std::stringstream stringStream;
   stringStream << duration / 1000;
-  std::string durationString = stringStream.str();
+  const std::string durationString = stringStream.str();
```
Comment (Kozoriz) [inline src/components/media_manager/src/audio/from_mic_to_file_recorder_thread.cc]: @LevchenkoS is there is correct naming?

Comment (LevchenkoS) [inline src/components/media_manager/src/audio/from_mic_to_file_recorder_thread.cc]: @Kozoriz, fixed in c5dcb82.

File: src/components/utils/src/file_system_posix.cc [hunk 2]
```diff
@@ -112,8 +112,9 @@ bool file_system::CreateDirectoryRecursively(const std::string& utf8_path) {
 
   while (ret_val == true && pos <= utf8_path.length()) {
     pos = utf8_path.find(GetPathDelimiter(), pos + 1);
+    const std::string path = utf8_path.substr(0, pos);
     if (!DirectoryExists(utf8_path.substr(0, pos))) {
-      if (0 != mkdir(utf8_path.substr(0, pos).c_str(), S_IRWXU)) {
+      if (0 != mkdir(path.c_str(), S_IRWXU)) {
```
Comment (wolfylambova22) [inline src/components/utils/src/file_system_posix.cc]: @LevchenkoS, so c.str() should remain here?

File: src/components/application_manager/src/commands/mobile/reset_global_properties_request.cc [hunk 3]
```diff
@@ -293,7 +292,7 @@ void ResetGlobalPropertiesRequest::on_event(const event_engine::Event& event) {
 
     SendResponse(result,
                  static_cast<mobile_apis::Result::eType>(result_code),
-                 return_info,
+                 return_info.c_str(),
```
Comment (VProdanov) [inline src/components/application_manager/src/commands/mobile/reset_global_properties_request.cc]: @LevchenkoS  return_info.empty() ? NULL : return_info.c_str()..

**Pull request merged**